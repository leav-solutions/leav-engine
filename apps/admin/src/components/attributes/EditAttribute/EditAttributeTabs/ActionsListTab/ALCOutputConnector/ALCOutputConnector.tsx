// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import styled from 'styled-components';
import ConnectorRect from '../ALCCard/ConnectorRect';
import {IColorDic} from '../interfaces/interfaces';

interface IALCOutputConnectorProps {
    connectionState: string | undefined;
    connColor: number[];
    size: number;
    types: string[];
    colorTypeDictionnary: IColorDic;
    forList?: boolean;
}

interface IBottomStyle {
    connectionState: string | undefined;
    forList?: boolean;
}

interface ITopTriangle {
    connColor: number[];
}

const BottomConnector = styled.div<IBottomStyle>`
    height: 20px;
    position: relative;
    bottom: 0;
    right: ${props => (props.forList ? 'auto' : '0')};
    top: ${props => (props.forList ? '-5px' : '0')};
    right: ${props => (props.forList ? '0' : 'auto')};
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-bottom: ${props => (props.connectionState && props.connectionState === 'neutral' ? '20px' : '5px')};
`;

const BottomTriangle = styled.div<ITopTriangle>`
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 0 20px 20px 0;
    border-color: transparent
        ${props =>
            props.connColor
                ? `rgb(${props.connColor[0]}, ${props.connColor[1]}, ${props.connColor[2]})`
                : 'transparent'}
        transparent transparent;
`;

function ALCOutputConnector({
    connectionState,
    connColor,
    size,
    types,
    colorTypeDictionnary,
    forList
}: IALCOutputConnectorProps): JSX.Element {
    const width = size ? size : 100;
    return (
        <BottomConnector connectionState={connectionState} forList={forList}>
            {connColor && <BottomTriangle connColor={connColor} />}
            <ConnectorRect size={width} types={types} colorTypeDictionnary={colorTypeDictionnary} />
        </BottomConnector>
    );
}

export default ALCOutputConnector;
