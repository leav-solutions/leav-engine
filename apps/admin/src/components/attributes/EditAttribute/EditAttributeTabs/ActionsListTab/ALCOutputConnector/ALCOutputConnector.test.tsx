// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {render} from 'enzyme';
import ALCOutputConnector from './ALCOutputConnector';

describe('ALCOutputConnector', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <ALCOutputConnector
                connectionState="connected"
                connColor={[]}
                size={100}
                types={['']}
                colorTypeDictionnary={{int: [255, 255, 255]}}
            />
        );

        expect(comp).toMatchSnapshot();
    });
});
