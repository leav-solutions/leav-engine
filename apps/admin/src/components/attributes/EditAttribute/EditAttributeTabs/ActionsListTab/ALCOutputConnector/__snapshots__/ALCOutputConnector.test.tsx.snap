// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ALCOutputConnector Snapshot test 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": {
      "class": "sc-guDMob hufcgN",
      "connectionstate": "connected",
    },
    "children": [
      Node {
        "attribs": {
          "class": "sc-dmyDmy iGzLSv",
          "conncolor": "",
        },
        "children": [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": {
            "class": "sc-beyTiQ brojHi",
            "style": "width:40px",
          },
          "children": [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": {
          "class": undefined,
          "conncolor": undefined,
        },
        "x-attribsPrefix": {
          "class": undefined,
          "conncolor": undefined,
        },
      },
      Node {
        "attribs": {
          "class": "sc-beyTiQ brojHi",
          "style": "width:40px",
        },
        "children": [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": {
            "class": "sc-dmyDmy iGzLSv",
            "conncolor": "",
          },
          "children": [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {
            "class": undefined,
            "conncolor": undefined,
          },
          "x-attribsPrefix": {
            "class": undefined,
            "conncolor": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": {
      "class": undefined,
      "connectionstate": undefined,
    },
    "x-attribsPrefix": {
      "class": undefined,
      "connectionstate": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": [
        Node {
          "attribs": {},
          "children": [
            Node {
              "attribs": {},
              "children": [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": {},
                "children": [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
            Node {
              "attribs": {},
              "children": [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": {},
                "children": [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {},
          "x-attribsPrefix": {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
